### DISCUSSION TITLE
combining useInfiniteQuery and usePaginatedQuery

### DISCUSSION THREAD

**Discussion by gffuma** [TanStack/query#734] [General] [answered]
Hi,
It would be cool having an hook that combining the infinite query mechanism from `useInfiniteQuery` and the `resolvedData` (The last successfully resolved data for the query) from `usePaginatedQuery`.
The uses case is to implement a filtrable infinite list:
(The code below is an example)

```js

const fetchProducts = (key, filters, offset) => {
  // Perform fetch ...
}

function ProductsList() {
  const [category, setCategory] = useState('foo')
  const { data } = useInfiniteQuery(
    ['products', { category }],
    fetchProducts,
    {
      getFetchMore: (lastGroup) => lastGroup.pagination.offset,
    }
  )
}
```
When i apply a new filter and the key changed i want to have similar to `usePaginatedQuery` the last value of data instead of null or causing suspending if the `suspense` mode is active.

**Comment by swandir** [comment]
Wouldn't it make sense to have this behavior by default? Seeing as infinite lists are the primary use case for useInfiniteQuery, and that such lists are often filtered.

**Comment by tannerlinsley** [comment, marked as answer]
This is a good idea. We'll do this in v3. I imagine that a "paginated" query will just be some type of modifier on the `useQuery` and `useInfiniteQuery` hooks that makes them lag the data behind for pagination :)

**Reply by tannerlinsley** [reply]
#772
